dts/bindings/video/video-interfaces.yaml

```diff
@@ -0,0 +1,148 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+# Common properties for video interface endpoints.
+
+include: base.yaml
```

Comment:
you had base.yaml included in the video interface binding, I don't think that was needed, you might be able to fix the build error and remove the need for the zephyr,video-interfaces compatible by remove the include: base.yaml , can you try it?

Reply:
@decsny : Indeed, the error comes from the included `base.yaml` which requires `compatible`. I misthought that I need it for `'#address-cells', '#size-cells' and 'reg' properties` when we have multiple ports (as described in the binding) but in fact, these are required for the port node, not for the endpoint node. So, removing the base.yaml resolves the issue.

I am going to change to your suggestion but I still don't understand which advantages of the "include" method instead of the "compatible" method ? while I see one direct "drawback" of the "include" method is that we need to make much more changes to use the new binding than the "compatible" method. For example, we need to change:

- All the 3 .yaml files : `nxp,imx-csi.yaml`,` nxp,mipi-csi2rx.yaml`, `ovti,ov5640.yaml` to include this binding
- The `nxp_rt11xx.dtsi` file to declare the required `remote-endpoint-label` property
```
		mipi_csi2rx: mipi_csi2rx@40810000 {
			compatible = "nxp,mipi-csi2rx";
			reg = <0x40810000 0x200>;
			status = "disabled";

			ports {
				#address-cells = <1>;
				#size-cells = <0>;

				port@0 {
					reg = <0>;
					mipi_csi2rx_ep_out: endpoint {
						remote-endpoint-label = "csi_ep_in";
					};
				};

				port@1 {
					reg = <1>;
				};
			};
		};
```
- Then, the `nxp_btb44_ov5640.overlay` file

While with the `compatible` method, we need to change only the `nxp_btb44_ov5640.overlay` file.

Is the reason just to be synced with Linux ?

Reply:
yes, the current goal of the zephyr project is to follow the standards of DT set in other projects, I honestly question its value at this point as much as you are, and even the previous DT maintainers had a big debate on their hands about this, but that's the currently stated design goal, and until the project decides otherwise, we should follow it's design goals including the one about being compatible with other projects like linux where possible. https://docs.zephyrproject.org/latest/build/dts/design.html#source-compatibility-with-other-operating-systems

And yes, you can find a lot of examples in zephyr where we are not being compatible, but this is because of either
a) the contributions were not reviewed properly (DT maintainers can only review so much, there are literally thousands of PR per month)
b) something was let slide because we literally could not find an alternative (for example, multi instance static software configuration is missing in zephyr and DT sort of works to do it but it's not a good design practice, arch WG has been finally very seriously discussing introducing a solution for this for the last few months)
c) zephyr DT tooling has some deficiency (I hope we can start to expand zephyr DT binding language, you have to remember zephyr had to write and still maintain it's own DT and binding compilers, we dont use DTC)

Personally, I think we are reaching a point with zephyr where we should just give up on source compatibility with other OS because so many things have slipped through. And it's very hard to get all the hundreds of contributors to understand the goals especially when the zephyr use case of DT is so strange and not fitting as well (remember the people who designed DT are also linux maintainers, I think zephyr is clearly showing that it doesnt work as well as project agnostic as they thought it was, but also, it was designed for totally different type of systems, not for generating C macros). But this is not a decision you and I are going to be making on this thread, it requires project level voting in the working groups and maybe even the TSC, and I'm not gonna argue about it any more here because it's pointless. Also I don't think the changes to the yaml files you mentioned are really that extensive, are they?

Reply:
> Also I don't think the changes to the yaml files you mentioned are really that extensive, are they?

No, they aren't. I mentioned just to make a comparison between the two methods as I didn't understand the reason. Thank you for these information !

Reply:
Very insightful!